contracts/utils/cryptography/P256.sol

```diff
@@ -171,10 +171,11 @@ library P256 {
     function _affineFromJacobian(uint256 jx, uint256 jy, uint256 jz) private view returns (uint256 ax, uint256 ay) {
         if (jz == 0) return (0, 0);
         uint256 zinv = Math.invModPrime(jz, P);
-        uint256 zzinv = mulmod(zinv, zinv, P);
-        uint256 zzzinv = mulmod(zzinv, zinv, P);
-        ax = mulmod(jx, zzinv, P);
-        ay = mulmod(jy, zzzinv, P);
+        assembly ("memory-safe") {
+            let zzinv := mulmod(zinv, zinv, P)
+            ax := mulmod(jx, zzinv, P)
+            ay := mulmod(jy, mulmod(zzinv, zinv, P), P)
```

Comment:
i think that `mulmod` acts the same within the assembly block, but need another pair of eyes to double check

Reply:
ah yes, great